packages/eslint-plugin/src/rules/array-type.ts

```diff
@@ -102,27 +102,30 @@ export default util.createRule<Options, MessageIds>({
       errorStringGenericSimple:
         "Array type using '{{readonlyPrefix}}{{type}}[]' is forbidden for non-simple types. Use '{{className}}<{{type}}>' instead.",
     },
-    schema: [
-      {
-        definitions: {
-          arrayOption: {
-            enum: ['array', 'generic', 'array-simple'],
-          },
+    schema: {
+      definitions: {
+        arrayOption: {
+          enum: ['array', 'generic', 'array-simple'],
         },
-        properties: {
-          default: {
-            $ref: '#/definitions/arrayOption',
-            description: 'The array type expected for mutable cases...',
-          },
-          readonly: {
-            $ref: '#/definitions/arrayOption',
-            description:
-              'The array type expected for readonly cases. If omitted, the value for `default` will be used.',
+      },
+      prefixItems: [
```

Comment:
This syntax is a little more heavyweight than `schema: [ ... ]`. It would be nice if someone could figure out a ray to use `$ref`s within the array. I couldn't.

cc @bchery 😁